### PR TITLE
CfP review system

### PR DESCRIPTION
**Pull request by SamLR**
Adds the 'cfp-review' blue print. This implements the CfP process workflow:

1. Submitted proposals that have been locked (by `make lockproposals`) are checked for general suitability (e.g. code of conduct violations) by an admin (under the cfp-review/proposals page) and hence sent to be anonymised
    * proposals can be rejected and/or futher information requested via the 'messages' system
2. Anonymisers can adjust the title and description on proposals
    * proposals that can't be anonymised are returned to the admin who can either request further information from the submitter or anonymise it themselves.
3. Anonymised proposals' titles and descriptions are then visible to reviewers who can either vote on a proposals, abstain from voting (if they have a conflict of interest) or ask further questions (sent via the admin using the 'notes' system).
4. When ready the admin can close a round by setting a threshold number of votes, proposals that reach this minimum number of votes (discounting abstentions and those with unresolved notes from reviewers) are marked as 'reviewed'
5. Each reviewed proposal has a score calculated for it by using the majority judgement system of voting
6. The admin then sets a threshold score for acceptance, accepted proposals are alerted and move to that state, those that aren't accepted are alerted the first time they fail to be accepted only.

This PR adds several horrible systems, 2 daft state machines and a load of pages. Sorry

### CODE REVIEW

File: templates/cfp_review/email/accepted_msg.txt [hunk 1]
```diff
@@ -0,0 +1,9 @@
+Hi {{ user.name }},
+
+Your {{ proposal.type }} ({{ proposal.title }}) has been accepted!
+We'll be in contact shortly to check how you'd like it to listed as well as to
```
Comment (marekventur) [inline templates/cfp_review/email/accepted_msg.txt]: Minor: I think there's a "be" missing in this sentence

File: templates/cfp_review/review_list.html [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+        <td><a href="{{ url_for('.review_proposal', proposal_id=proposal.id) }}">{{proposal.title}}</a></td>
+        <td>{% if proposal.category %}
+            {{ proposal.category.name }}
+        {% else %}
+            --
+        {% endif %}</td>
+        <td>{{ proposal.description | truncate(100) }}</td>
+    </tr>
+{% else %}
+    <td colspan="3" class="text-center">No proposals to review</td>
```
Comment (marekventur) [inline templates/cfp_review/review_list.html]: Minor: I think this should be wrapped in a <tr> and have a colspan=4

File: templates/cfp_review/review_list.html [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+        <td>
+            {% if vote.note %}
+                {{ vote.note | truncate(300) }}
+            {% else %}
+                    --
+            {% endif %}
+        </td>
+    </tr>
+{% else %}
+    <td colspan="5" class="text-center">No reviewed proposals</td>
```
Comment (marekventur) [inline templates/cfp_review/review_list.html]: Minor: I think this also needs to be wrapped in a <tr>

File: templates/cfp_review/email/accepted_msg.txt (hunk 1 above)
Comment (SamLR) [inline templates/cfp_review/email/accepted_msg.txt]: fixed in 61de36b

File: templates/cfp_review/review_list.html (hunk 3 above)
Comment (SamLR) [inline templates/cfp_review/review_list.html]: fixed in b81354b

**Comment by marksteward** [comment]
:+1: ;)

**Pull request merged**